examples/text/amazonreviewpolarity.py

```diff
@@ -37,25 +37,13 @@ def AmazonReviewPolarity(root, split):
     """Demonstrating caching, extraction and sanity check pipelines."""
 
     url_dp = IterableWrapper([URL])
-    # cache data on-disk with sanity check
-    cache_dp = url_dp.on_disk_cache(
+    cache_compressed_dp = url_dp.on_disk_cache(
         filepath_fn=lambda x: os.path.join(root, _PATH), hash_dict={os.path.join(root, _PATH): MD5}, hash_type="md5"
     )
-    cache_dp = GDriveReader(cache_dp).end_caching(mode="wb", same_filepath_fn=True)
-
-    cache_dp = FileOpener(cache_dp, mode="b")
-
-    # stack TAR extractor on top of loader DP
-    extracted_files = cache_dp.read_from_tar()
-
-    # filter files as necessary
-    filter_extracted_files = extracted_files.filter(lambda x: split in x[0])
-
-    # stack sanity checker on top of extracted files
-    check_filter_extracted_files = filter_extracted_files.check_hash(
-        {os.path.normpath(os.path.join(root, _EXTRACTED_FILES[split])): _EXTRACTED_FILES_MD5[split]},
-        "md5",
-    )
-
-    # stack CSV reader and do some mapping
-    return check_filter_extracted_files.parse_csv().map(fn=lambda t: (int(t[0]), t[1]))
+    cache_compressed_dp = GDriveReader(cache_compressed_dp).end_caching(mode="wb", same_filepath_fn=True)
+    cache_decompressed_dp = cache_compressed_dp.on_disk_cache(
+        filepath_fn=lambda x: os.path.join(root, os.path.dirname(_EXTRACTED_FILES[split]), os.path.basename(x)))
+    cache_decompressed_dp = FileOpener(cache_decompressed_dp, mode="b").read_from_tar()
+    cache_compressed_dp = cache_decompressed_dp.end_caching(mode="wb", same_filepath_fn=True)
+    data_dp = FileOpener(cache_decompressed_dp.filter(lambda x: _EXTRACTED_FILES[split] in x[0]).map(lambda x: x[0]), mode='b')
```

Comment:
Hi @ejguan, it seems like the end_caching for decompressed files are not saved onto the disk. I am getting following error: 
`NotADirectoryError: [Errno 20] Not a directory: '/Users/parmeetbhatia/.torchtext/cache/AmazonReviewPolarity/amazon_review_polarity_csv.tar.gz/amazon_review_polarity_csv/train.csv'` 

not sure, what i am doing wrong here. could you please help investigate this?

Reply:
After `read_from_tar`, the data becomes "(decompressed file path, file handle)" where decompressed file path becomes `amazon_review_polarity_csv/amazon_review_polarity_csv/train.csv`. But, as you used `same_filepath_fn=True`, the same function from `on_disk_cache` would be applied to each data, then you would get each file to `amazon_review_polarity_csv.tar.gz/amazon_review_polarity_csv/train.csv`.

You can use `end_caching(mode="wb")` without specifying `same_filepath_fn` to get `amazon_review_polarity_csv/amazon_review_polarity_csv/train.csv` for each decompressed file.